src/parser/shaman/restoration/modules/azerite/SoothingWaters.js

```diff
@@ -0,0 +1,84 @@
+import SPELLS from 'common/SPELLS';
+import StatTracker from 'parser/shared/modules/StatTracker';
+import CritEffectBonus from 'parser/shared/modules/helpers/CritEffectBonus';
+import BaseHealerAzerite from './BaseHealerAzerite';
+
+const HEAL_WINDOW_MS = 100;
+
+/**
+ * Soothing Waters:
+ * Your Chain Heal heals its primary target for an additional 756 healing.
+ */
+
+class SoothingWaters extends BaseHealerAzerite {
+  static dependencies = {
+    statTracker: StatTracker,
+    critEffectBonus: CritEffectBonus,
+  };
+  static TRAIT = SPELLS.SOOTHING_WATERS_TRAIT.id;
+  static HEAL = SPELLS.SOOTHING_WATERS_TRAIT.id;
+
+  traitRawHealing = 0;
+
+  chainHealEvent;
+  chainHealTarget;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.hasTrait;
+    if (!this.active) {
+      return;
+    }
+
+    this.traitRawHealing = this.azerite.reduce((total, trait) => total + trait.rawHealing, 0);
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+
+    if (spellId !== SPELLS.CHAIN_HEAL.id) {
+      return;
+    }
+
+    // filtering out healing events if they're not from the current cast
+    if (this.chainHealEvent && event.timestamp - this.chainHealEvent.timestamp >= HEAL_WINDOW_MS) {
+      this.chainHealEvent = undefined;
```

Comment:
I'd personally use null here (and also when the variable is declared) instead of undefined, but idk if we have a standard for that or if both are fine.